core/src/sumeragi/fault.rs

```diff
@@ -68,194 +66,55 @@ where
     pub(crate) transaction_validator: TransactionValidator,
     pub(crate) telemetry_started: bool,
-    /// Genesis network
-    pub genesis_network: Option<GenesisNetwork>,
     /// Broker
     pub broker: Broker,
     /// Kura instance used for IO
     pub kura: Arc<Kura>,
     /// [`iroha_p2p::Network`] actor address
-    // pub network: Addr<IrohaNetwork>,
+    pub network: Addr<IrohaNetwork>,
     /// Buffer capacity of actor's MPSC channel
     pub actor_channel_capacity: u32,
     pub(crate) fault_injection: PhantomData<F>,
     pub(crate) gossip_batch_size: u32,
     pub(crate) gossip_period: Duration,
+
+    pub atomic_sumeragi_state: Mutex<AtomicSumeragiState>,
+    pub current_online_peers_by_public_key: Mutex<Vec<PublicKey>>,
+
+    pub incomming_message_sender: Mutex<std::sync::mpsc::Sender<Message>>,
+    pub incomming_message_receiver: Mutex<std::sync::mpsc::Receiver<Message>>,
 }
 
-impl<F: FaultInjection> SumeragiWithFault<F> {
-    fn from_configuration(
-        configuration: &SumeragiConfiguration,
-        events_sender: EventsSender,
-        wsv: WorldStateView,
-        transaction_validator: TransactionValidator,
-        telemetry_started: bool,
-        genesis_network: Option<GenesisNetwork>,
-        queue: Arc<Queue>,
-        broker: Broker,
-        kura: Arc<Kura>,
-        // network: Addr<IrohaNetwork>,
-    ) -> Result<Self> {
-        let network_topology = Topology::builder()
-            .at_block(EmptyChainHash::default().into())
-            .with_peers(configuration.trusted_peers.peers.clone())
-            .build()?;
-
-        Ok(Self {
-            key_pair: configuration.key_pair.clone(),
-            topology: network_topology,
-            peer_id: configuration.peer_id.clone(),
-            voting_block: None,
-            votes_for_blocks: BTreeMap::new(),
-            events_sender,
-            wsv: std::sync::Mutex::new(wsv),
-            txs_awaiting_receipts: HashMap::new(),
-            txs_awaiting_created_block: HashSet::new(),
-            votes_for_view_change: HashMap::new(),
-            commit_time: Duration::from_millis(configuration.commit_time_limit_ms),
-            tx_receipt_time: Duration::from_millis(configuration.tx_receipt_time_limit_ms),
-            block_time: Duration::from_millis(configuration.block_time_ms),
-            block_height: 0,
-            invalidated_blocks_hashes: Vec::new(),
-            telemetry_started,
-            transaction_limits: configuration.transaction_limits,
-            transaction_validator,
-            genesis_network,
-            queue,
-            broker,
-            kura,
-            // network,
-            actor_channel_capacity: configuration.actor_channel_capacity,
-            fault_injection: PhantomData,
-            gossip_batch_size: configuration.gossip_batch_size,
-            gossip_period: Duration::from_millis(configuration.gossip_period_ms),
-        })
-    }
+pub struct AtomicSumeragiState {
```

Comment:
It has the word `Atomic` in it. But nothing so far suggests to me that it is `Atomic`. Moreover, a struct made up of atomic units isn't itself atomic. I know that this is WIP, but we need 0day good naming conventions.

Reply:
I've changed it to SumeragiStateMachineData. Good of you two for pushing me on this.